src/Access/EnabledQuota.cpp

```diff
@@ -67,7 +67,7 @@ struct EnabledQuota::Impl
             {
                 /// We reset counters only if the interval's end has been calculated before.
                 /// If it hasn't we just calculate the interval's end for the first time and don't reset counters yet.
-                need_reset_counters = (end_loaded.count() != 0);
+                need_reset_counters = (end_of_interval.load().count() != 0);
```

Comment:
I finally understand the problem you want to fix.

Let's change this line to a simplier equivalent

```
need_reset_counters = true;
```

and in order to fix the tests in more correct way please add the following lines:

```
if (!interval.end_of_interval.load().count())
{
    /// We need to calculate end of the interval if it hasn't been calculated before.
    bool dummy;
    getEndOfInterval(interval, current_time, dummy);
}
```

before [this](https://github.com/ClickHouse/ClickHouse/blob/9892151290bee34f068be82355a8c0b54e67fb88/src/Access/EnabledQuota.cpp#L98) line and also before [this](https://github.com/ClickHouse/ClickHouse/blob/9892151290bee34f068be82355a8c0b54e67fb88/src/Access/EnabledQuota.cpp#L126).

Reply:
ok